app/controllers/api/v1/twilio_controller.rb

```diff
@@ -3,13 +3,47 @@
 
 module Api::V1
   class TwilioController < Api::V1::ApiController
-    include Webhookable
+    include TwilioConfig
 
-    after_filter :set_header, except: [:assignment, :hold_music]
+    after_filter :set_header, except: [:assignment, :hold_music, :twilio_generate_call_token]
     skip_authorization_check
     skip_before_action :validate_token
     skip_before_action :verify_authenticity_token
 
+    # OUTBOUND CALL : START
+    def connect_outbound_call
+      caller_id, offer_id, mobile = params["phone_number"].split("#")
+      redis.hmset("twilio_outbound_#{mobile}",
+        :offer_id, offer_id,
+        :caller_id, caller_id)
+
+      response = Twilio::TwiML::Response.new do |r|
+        r.Dial callerId: voice_number, action: api_v1_twilio_completed_outbound_call_path do |d|
+          d.Number mobile
+        end
+      end
+      render_twiml response
+    end
+
+    def completed_outbound_call
+      response = Twilio::TwiML::Response.new do |r|
+        unless params["DialCallStatus"] == "completed"
+          r.Say "Couldn't reach #{user(child_call.to).full_name} try again soon. Goodbye."
+        end
+        r.Hangup
+      end
+      render_twiml response
+    end
+
+    def outbound_call_status
+      offer_id = redis.hmget("twilio_outbound_#{child_call.to}", :offer_id)
+      user_id  = redis.hmget("twilio_outbound_#{child_call.to}", :caller_id)
```

Comment:
@swatijadhav As per comment above, it would be good to extract hmget into a service class. In this particular case it would mean you could make one redis call and extract both params rather than having to call redis twice here.

    tcm = TwilioCallManager.new(to: child_call.to)
    tcm.offer_id
    tcm.user_id

and

    class TwilioCallManager
    
      OutBoundCallPrefix = "twilio_outbound_"
      
      def initialize(options={})
        @call_to = options[:to]
      end
      
      def offer_id
        data[:offer_id]
      end
      
      def user_id
        data[:user_id]
      end
      
      private
      
      def data
        @data ||= redis.hmget("#{OutBoundCallPrefix}_#{@to}")
      end
      
      def redis
        @redis ||= Goodcity::Redis.new
      end
      
    end